packages/shared/src/index.ts

```diff
@@ -51,6 +62,14 @@ export const hasOwn = (
 ): key is keyof typeof val => hasOwnProperty.call(val, key)
 
 export const isArray = Array.isArray
+export const isArrayLike = (val: unknown): val is ArrayLike<any> => {
+  try {
+    ;[...(val as any)]
```

Comment:
This solution would exhaust the iteration of any one-shot iterators. (Though I'm unsure if Vue actually supports using these in reactive data). It could likely also have unnecessary perf overhead.

I think a better definition of array-like would be 1) Having a length property or 2) Having a defined `Symbol.Iterator`.

Reply:
If this would impact the overall performance, plus I don't see the urge requirements to support TypedArray, it's may not worthy to deal with this.